src/algorithms/synthesis/encoding.cpp

```diff
@@ -2,10 +2,44 @@
 
 namespace syrec {
 
-    auto encodeHuffman(TruthTable& tt) -> TruthTable::CubeMap {
-        std::map<TruthTable::Cube, std::size_t> outputFreq;
+    auto encodeHuffman(TruthTable& tt, bool additionalLine) -> TruthTable::CubeMultiMap {
```

Comment:
This function is getting way too complicated. There should definitely be two functions for encoding with or without additional lines. Common functionality should be externalised into separate functions.